README.md

```diff
@@ -1,12 +1,126 @@
+# Official SDF GitHub Action
+
+This action is used to run SDF in CI/CD workflows with GitHub Actions. A common use case would be to run an impact analysis with `sdf compile` and `sdf check` on PR creation. 
+
+*Note: This GitHub action is still < v1, as such certain scenarios may result in unexpected behavior. Please follow the [contributing guidelines](./CONTRIBUTING.md) and create an issue in this repo if you find any bugs or issues.*
+
+For an in-depth guide on how to use this GitHub, please see the GitHub CI/CD section of [our official docs](https://docs.sdf.com/integrations/cicd/ci_cd)
 
 ## Usage
 Check out this [example workflow](./.github/workflows/examples.yml) to see how to use this action.
 
+<!-- start usage -->
+```yaml
+- uses: sdf-labs/sdf-action@v0
+  with:
+    # The version of SDF to use in the GitHub action.
+    # This should match the version of SDF you use locally in development.
+    sdf_version: '0.2.21'
+
+    # The SDF command to run. The exit code of this command is used to determine
+    # success or failure of the action.
+    command: 'sdf compile'
+
+    # The relative path to the workspace starting from the root of your repository.
+    # The workspace directory is where the `workspace.sdf.yml` file is located.
+    workspace_dir: './my_sdf_workspace/'
+
+    # Whether or not this SDF project is being used with dbt. If so, 
+    # the action runs `sdf dbt refresh` in the container before the 
+    # specified SDF command.
+    is_dbt: 'false' # Defaults to false
+
+    ### SDF Cloud Authentication
+   
+    # NOTE: Both the access_key and secret_key parameters are required for accessing the SDF 
+
+    # The access key used for headless authentication to the SDF Cloud
+    access_key: ${{ secrets.SDF_ACCESS_KEY }}
+
+    # The secret key used for headless authentication to the SDF Cloud
+    secret_key: ${{ secrets.SDF_SECRET_KEY }}
+
+    ### Snowflake Integration
+
+    # All of the following are used to authenticate to Snowflake.
+    # All of these are required if using a Snowflake provider.
+    snowflake_account_id: ${{ secrets.SNOWFLAKE_ACCOUNT_ID }}
+    snowflake_username: ${{ secrets.SNOWFLAKE_USERNAME }}
+    snowflake_password: ${{ secrets.SNOWFLAKE_PASSWORD }}
+    snowflake_role: ${{ secrets.SNOWFLAKE_ROLE }}
+    snowflake_warehouse: ${{ secrets.SNOWFLAKE_WAREHOUSE }}
+
+    ### AWS / Redshift Integration
+
+    # All of the following are required if using an S3, AWS Glue, or AWS Redshift
+    # provider.
+    aws_region: ${{ secrets.AWS_REGION }}
+    aws_access_key_id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+    aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+```
+<!-- end usage -->
+
+# Scenarios
+
+[Run Compile on PR](#run-compile-on-pr-creation-and-update)
+[Run Checks on PR with Snowflake Provider](#run-compile-on-pr-creation-and-update-with-a-snowflake-provider)
+[Push to SDF Cloud on Merge to Main](#push-to-sdf-cloud-on-merge-to-main-with-a-redshift-provider)
```

Comment:
nit: these 3 are on 1 line, is this intended?

Reply:
Just updated to make it more clear. The VSCOde table of contents auto gen was taking over when I tried to make them a list. Very annoying